### PR TITLE
Heterogenous Agent Computing

### PR DESCRIPTION
**Pull request by davidrusu**
This PR contains a first pass at Heterogeneous Agent Computing. 

The main idea is that we have the Zoo Worker installed on a server and listens for SMARTS sessions requesting remote social agents.

```bash
# on a server
scl zoo worker

# locally

gym.make(..., zoo_workers=[("143.125.34.1", 1200)])
```

Sequence diagram for SMARTS allocating a remote social agent
zoo-compute-sequence-diagram

### CODE REVIEW

**Review by Adaickalavan — APPROVED** [review]

File: envision/web/src/index.js [hunk 1]
```diff
@@ -26,7 +26,7 @@ import App from "./components/app.js";
 import "antd/dist/antd.dark.css";
 
 let client = new Client({
-  endpoint: "http://localhost:8081",
+  endpoint: window.location.origin,
```
Comment (JVillella) [inline envision/web/src/index.js:29]: This includes the port I take it.

**Review by JVillella — APPROVED** [review]
I like it, thanks for the review last night on the design. Perhaps attach the diagram to this PR for record keeping (could also put it in the wiki)?

**Review by liamchzh — APPROVED** [review]

**Comment by liamchzh** [comment]
A giant step on HAC!

File: smarts/zoo/worker.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        sock_file = tempfile.mktemp()
+        proc = spawn_local_agent(sock_file)
+        return proc, {"socket_file": sock_file, "result": "success"}
+
+    else:
+        return None, {"result": "error", "msg": "bad request"}
+
+
+def listen(port):
+    logging.basicConfig(level=logging.INFO)
```
Comment (Gamenot) [inline smarts/zoo/worker.py:53]: I think this is not the place for setting the logging level.

**Review by Gamenot — APPROVED** [review]

File: envision/web/src/index.js (hunk 1 above)
Comment (davidrusu) [reply envision/web/src/index.js:29]: yep, sure does

File: examples/single_agent.py [hunk 3]
```diff
@@ -36,6 +36,7 @@ def main(scenarios, headless, num_episodes, seed, max_episode_steps=None):
         timestep_sec=0.1,
         sumo_headless=True,
         seed=seed,
+        # zoo_workers=[("143.110.210.157", 7432)],
```
[the remark this answers is not in the document]
Comment (davidrusu) [reply examples/single_agent.py:39]: Added a comment

File: smarts/zoo/worker.py (hunk 2 above)
Comment (davidrusu) [reply smarts/zoo/worker.py:53]: Moved it to the global scope

**Pull request merged**